File: ArtGalleryManagementSystemAPI/ArtGalleryManagementSystemAPI/Services/ProductService.cs

using ArtGalleryManagementSystemAPI.Dtos;

namespace ArtGalleryManagementSystemAPI.Services;

public interface ProductService
{
    public List<ProductDto> FindAll();
    public List<ProductDto> FindBySellerId(int id);
    public ProductWithAttributesDto FindById(int id);
    public ProductWithSellerDto FindByIdWithSeller(int id);
    public ProductWithAttributesDto FindByIdWithAttributes(int id);
    public List<ProductWithSellerDto> FindByCategoryId(int id);
    public List<ProductWithSellerDto> AllProductWithSeller();
    public List<ProductWithSellerDto> SortByPrice(double min, double max);
    public List<ProductWithSellerDto> SortByPriceLowHigh(string value, double min, double max);
    public List<ProductWithSellerDto> SearchByKeyword(string value);
    public List<CategoryDto> FindAllCategory();
    public bool PostArt(ProductWithAttributesDto productWithAttributesDto);

    public List<ProductWithSellerDto> AllAuctionProductWithSeller();

    public bool EditArt(ProductWithAttributesDto productWithAttributesDto);

    public bool AddReview(ReviewDto reviewDto);
    public List<ReviewDto> FindAllReview();
    public List<ReviewDto> FindAllReviewByProId(int proId);

    public ReviewDto FindReviewByProId(int proId, int userId, string createdAt);
    public List<ReviewDto> FindReviewByUserId(int userId);
    public bool Delete(ProductDto productDto);


}

File: ArtGalleryManagementSystemUser/src/app/user/services/product.service.ts

import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { lastValueFrom } from "rxjs";
import { BaseURLService } from "./baseURL.service";

@Injectable({
    providedIn: 'root'
})
export class ProductService{
    constructor(
        private httpClient: HttpClient,
        private baseUrlService : BaseURLService
    ){}
    async findall(){
        return lastValueFrom(this.httpClient.get(this.baseUrlService.BASE_URL+'home/findallproduct'))
    }
    async findallwithseller(){
        return lastValueFrom(this.httpClient.get(this.baseUrlService.BASE_URL+'home/findallproductwithseller'))
    }
    async sortbyprice(min:number, max:number){
        return lastValueFrom(this.httpClient.post(this.baseUrlService.BASE_URL+'home/sortbyprice',{min,max}))
    }
    async sortbypricelowhigh(value:string, min:number, max:number){
        return lastValueFrom(this.httpClient.post(this.baseUrlService.BASE_URL+'home/sortbypricelowhigh',{value,min,max}))
    }
    async searchByKeyword(value:string){
        return lastValueFrom(this.httpClient.get(this.baseUrlService.BASE_URL+'home/searchbykeyword/'+value))
    }
    async findProductId(productId:number){
        return lastValueFrom(this.httpClient.get(this.baseUrlService.BASE_URL+'home/findproductbyid/'+productId))
    }
    async findProductIdWithSeller(productId:number){
        return lastValueFrom(this.httpClient.get(this.baseUrlService.BASE_URL+'home/findproductbyidwithseller/'+productId))
    }
    async findProductIdWithAttributes(productId:number){
        return lastValueFrom(this.httpClient.get(this.baseUrlService.BASE_URL+'home/findproductbyidwithattributes/'+productId))
    }
    async postart(formData:FormData){
        return lastValueFrom(this.httpClient.post(this.baseUrlService.BASE_URL+'home/createrart',formData))
    }
    async editart(formData:FormData){
        return lastValueFrom(this.httpClient.put(this.baseUrlService.BASE_URL+'home/editart',formData))
    }
    async findallcategory(){
        return lastValueFrom(this.httpClient.get(this.baseUrlService.BASE_URL+'home/findallcategory'))
    }
    async findbycategoryid(id:number){
        return lastValueFrom(this.httpClient.get(this.baseUrlService.BASE_URL+'home/findbycategoryid/'+id))
    }
    async findallbyseller(id:number){
        return lastValueFrom(this.httpClient.get(this.baseUrlService.BASE_URL+'home/findallbyseller/'+id))
        
    }
    async findbyid(id:number){
        return lastValueFrom(this.httpClient.get(this.baseUrlService.BASE_URL+'home/findproductbyid/'+id))
    }
    async addreview(formdata:any){
        return lastValueFrom(this.httpClient.post(this.baseUrlService.BASE_URL+'home/addreview',formdata))
    }
    async findreviewbyproId(proId:number,userId:number,createdAt:string){
        return lastValueFrom(this.httpClient.post(this.baseUrlService.BASE_URL+'home/findreviewbyproId',{proId,userId,createdAt}))
    }
    async findreviewbyuserid(userId:number){
        return lastValueFrom(this.httpClient.get(this.baseUrlService.BASE_URL+'home/findreviewbyuserid/'+userId))
    }
    async findallreview(){
        return lastValueFrom(this.httpClient.get(this.baseUrlService.BASE_URL+'home/findallreview'))
    }
    async findallreviewbyproid(proId:number){
        return lastValueFrom(this.httpClient.get(this.baseUrlService.BASE_URL+'home/findallreviewbyproid/'+proId))
    }
    async deleteproduct(formData:FormData){
        return lastValueFrom(this.httpClient.put(this.baseUrlService.BASE_URL + 'home/delete', formData));
    }
}